src/AppBundle/Controller/BlogController.php

```diff
@@ -150,4 +151,42 @@ public function commentFormAction(Post $post)
             'form' => $form->createView(),
         ]);
     }
+
+    /**
+     * @Route("/search", name="blog_search")
+     * @Method("GET")
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $query = $request->query->get('q', '');
+
+        // Sanitizing the query: removes all non-alphanumeric characters except whitespaces
+        $query = preg_replace('/[^[:alnum:] ]/', '', trim(preg_replace('/[[:space:]]+/', ' ', $query)));
+
+        // Splits the query into terms and removes all terms which
+        // length is less than 2
+        $terms = array_unique(explode(' ', mb_strtolower($query)));
+        $terms = array_filter($terms, function ($term) {
+            return 2 <= mb_strlen($term);
+        });
+
+        $posts = [];
+
+        if (!empty($terms)) {
+            $posts = $this->getDoctrine()->getRepository(Post::class)->findByTerms($terms);
+        }
+
+        $results = [];
+
+        foreach ($posts as $post) {
+            array_push($results, [
```

Comment:
We finally used `$results[] = xxx` ... mostly because PHPStorm displayed a pretty annoying message suggesting that 😄